WordPress/src/main/java/org/wordpress/android/networking/GlideRequestFactory.kt

```diff
@@ -0,0 +1,58 @@
+package org.wordpress.android.networking
+
+import android.util.Base64
+import com.android.volley.Request
+import com.android.volley.Request.Priority
+import com.bumptech.glide.integration.volley.VolleyRequestFactory
+import com.bumptech.glide.integration.volley.VolleyStreamFetcher
+import com.bumptech.glide.load.data.DataFetcher.DataCallback
+import org.wordpress.android.fluxc.network.HTTPAuthManager
+import org.wordpress.android.fluxc.network.UserAgent
+import org.wordpress.android.fluxc.network.rest.wpcom.auth.AccessToken
+import org.wordpress.android.util.UrlUtils
+import org.wordpress.android.util.WPUrlUtils
+import java.io.InputStream
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * RequestFactory which adds authorization headers to all Glide requests and makes sure requests to WPcom endpoints
+ * use https.
+ */
+@Singleton
+class GlideRequestFactory @Inject constructor(
+    private val accessToken: AccessToken,
+    private val httpAuthManager: HTTPAuthManager,
+    private val userAgent: UserAgent
+) : VolleyRequestFactory {
+    override fun create(
+        url: String,
+        callback: DataCallback<in InputStream>,
+        priority: Priority,
+        headers: Map<String, String>
+    ): Request<ByteArray>? {
+        val httpsUrl: String = convertWPcomUrlToHttps(url)
+        return VolleyStreamFetcher.GlideRequest(httpsUrl, callback, priority, addAuthHeaders(url, headers))
+    }
+
+    private fun convertWPcomUrlToHttps(url: String): String {
+        return if (WPUrlUtils.isWordPressCom(url) && !UrlUtils.isHttps(url)) UrlUtils.makeHttps(url) else url
+    }
+
+    private fun addAuthHeaders(url: String, currentHeaders: Map<String, String>): MutableMap<String, String> {
+        val headers = currentHeaders.toMutableMap()
+        headers.put("User-Agent", userAgent.userAgent)
+        if (WPUrlUtils.safeToAddWordPressComAuthToken(url)) {
+            headers.put("Authorization", "Bearer " + accessToken.get())
```

Comment:
We should validate whether or not we have a token here first - otherwise, requests to images on wordpress.com (e.g., the site avatar) attach this header:

```
authorization: Bearer
```

This doesn't cause any problems at the moment, but it looks weird and might cause us trouble down the road.